ibm/service/cloudshell/resource_ibm_cloud_shell_account_settings.go

```diff
@@ -146,209 +183,227 @@ func resourceIBMCloudShellAccountSettingsCreate(context context.Context, d *sche
 	}
 	if _, ok := d.GetOk("features"); ok {
 		var features []ibmcloudshellv1.Feature
-		for _, e := range d.Get("features").([]interface{}) {
-			value := e.(map[string]interface{})
-			featuresItem := resourceIBMCloudShellAccountSettingsMapToFeature(value)
-			features = append(features, featuresItem)
+		for _, v := range d.Get("features").([]interface{}) {
+			value := v.(map[string]interface{})
+			featuresItem, err := ResourceIBMCloudShellAccountSettingsMapToFeature(value)
+			if err != nil {
+				return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_cloud_shell_account_settings", "delete", "parse-features").GetDiag()
+			}
+			features = append(features, *featuresItem)
 		}
 		updateAccountSettingsOptions.SetFeatures(features)
 	}
 	if _, ok := d.GetOk("regions"); ok {
 		var regions []ibmcloudshellv1.RegionSetting
-		for _, e := range d.Get("regions").([]interface{}) {
-			value := e.(map[string]interface{})
-			regionsItem := resourceIBMCloudShellAccountSettingsMapToRegionSetting(value)
-			regions = append(regions, regionsItem)
+		for _, v := range d.Get("regions").([]interface{}) {
+			value := v.(map[string]interface{})
+			regionsItem, err := ResourceIBMCloudShellAccountSettingsMapToRegionSetting(value)
+			if err != nil {
+				return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_cloud_shell_account_settings", "delete", "parse-regions").GetDiag()
+			}
+			regions = append(regions, *regionsItem)
 		}
 		updateAccountSettingsOptions.SetRegions(regions)
 	}
 
-	accountSettings, response, err := ibmCloudShellClient.UpdateAccountSettingsWithContext(context, updateAccountSettingsOptions)
-	if err != nil || accountSettings == nil {
-		log.Printf("[DEBUG] UpdateAccountSettingsWithContext failed %s\n%s", err, response)
-		return diag.FromErr(fmt.Errorf("UpdateAccountSettingsWithContext failed %s\n%s", err, response))
+	accountSettings, _, err := ibmCloudShellClient.UpdateAccountSettingsWithContext(context, updateAccountSettingsOptions)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("UpdateAccountSettingsWithContext failed: %s", err.Error()), "ibm_cloud_shell_account_settings", "delete")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
 	}
 
-	d.SetId(*accountSettings.ID)
+	d.SetId(fmt.Sprintf("%s/%s", *updateAccountSettingsOptions.AccountID, *accountSettings.AccountID))
```

Comment:
This change will break all the existing user's 
To test this behavior
Use some latest released version to provision or create the resource
then use local build provider and run plan or apply it breaks